kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/ProduceActionIntegrationTest.java

```diff
@@ -3721,6 +3768,111 @@ public void produceBinaryWithLargerSizeMessage() throws Exception {
     assertEquals(Arrays.toString(value), Arrays.toString(produced.value()));
   }
 
+  private void doByteLimitReachedTest() throws Exception {
+    String clusterId = testEnv.kafkaCluster().getClusterId();
+    String key = "foo";
+    String value = "bar";
+    ProduceRequest request =
+        ProduceRequest.builder()
+            .setKey(
+                ProduceRequestData.builder()
+                    .setFormat(EmbeddedFormat.JSON)
+                    .setData(TextNode.valueOf(key))
+                    .build())
+            .setValue(
+                ProduceRequestData.builder()
+                    .setFormat(EmbeddedFormat.JSON)
+                    .setData(TextNode.valueOf(value))
+                    .build())
+            .setOriginalSize(0L)
```

Comment:
For curiosity what does this `setOriginalSize` do? Does this mean this request is 0 byte?

Reply:
Unfortunately if not set, AutoValue throws complaining. There was an intentional PR to make `originalSize` as non-optional https://github.com/confluentinc/kafka-rest/pull/941/files. I think it makes sense, to make sure originalSize is always set correctly, as it is used for billing metrics.

`originalSize` is meaningless when creating the request on the client(also note "actual" client won't use this java class to represent ProduceRequest, as originalSize isn't part of openAPI spec). Infact "true" originalSize is set server side, when ProduceRequest object is built during DeSer
https://github.com/confluentinc/kafka-rest/blob/f19d384487747493d9211fe9c6273b45c554821d/kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ProduceRequest.java#L130